### ISSUE TITLE
Chef/Modernize/CronDFileOrTemplate fails to match when the path property is used to provide the file path

### ISSUE THREAD

**Issue by yacn** [chef/cookstyle#943]
# Version:

7.31.3

# Environment:

* OS: Fedora 35
* cookbook: minimum required to reproduce issue

# Scenario:

cookstyle should flag when a `file`, `cookbook_file`, or `template` is used to manage a file in `'/etc/cron.d` when the file path is provided to the resource via the `path` property.

# Steps to Reproduce:

* run cookstyle on the following recipe:

```ruby
file 'delete old cron job' do
  path '/etc/cron.d/backup'
  action :delete
end
```

# Expected Result:

cookstyle should flag the above resource to be replaced with `cron_d`.

```
❯ ../bundlebin/cookstyle --only Chef/Modernize/CronDFileOrTemplate
Inspecting 2 files
.R

Offenses:

recipes/tst.rb:1:1: R: Chef/Modernize/CronDFileOrTemplate: Use the cron_d resource that ships with Chef Infra Client 14.4+ instead of manually creating the file with template, file, or cookbook_file resources (https://docs.chef.io/workstation/cookstyle/chef_modernize_crondfileortemplate)
file 'delete old job' do ...
```

# Actual Result:

No linting offenses raised

```
❯ cinc exec cookstyle .
Inspecting 2 files
..

2 files inspected, no offenses detected
```
